pkg/api/controller_test.go

```diff
@@ -3463,3 +3463,132 @@ func TestController_ClientDisconnect(t *testing.T) {
 		t.Fatalf("Metric for client request closed: %d, expected: %d", clientRequestClosedCount, expectedCount)
 	}
 }
+
+func TestController_SendStatsEvents(t *testing.T) {
+	clt, deps := setupClientWithAdmin(t)
+	ctx := context.Background()
+
+	tests := []struct {
+		name                string
+		events              []api.StatsEvent
+		expectedEventCounts []int
+		expectedStatusCode  int
+	}{
+		{
+			name: "single_event_count_1",
+			events: []api.StatsEvent{
+				{
+					Class: "single_event_count_1",
+					Name:  "name",
+					Count: 1,
+				},
+			},
+			expectedEventCounts: []int{1},
+			expectedStatusCode:  http.StatusNoContent,
+		},
+		{
+			name: "single_event_count_gt_1",
+			events: []api.StatsEvent{
+				{
+					Class: "single_event_count_gt_1",
+					Name:  "name",
+					Count: 3,
+				},
+			},
+			expectedEventCounts: []int{3},
+			expectedStatusCode:  http.StatusNoContent,
+		},
+		{
+			name: "multiple_events",
```

Comment:
Multiple events can hold the same class/name - we need to test it.
In order to test it I think you want to describe the expected events count as a map and not an array - we like to count the stats collector after posting the event and match the map that sums the total of each event type (class/name) to the expected count.

Reply:
Added tests and modified structures accordingly